conda_build/metadata.py

```diff
@@ -269,9 +269,21 @@ def get_contents(meta_path):
     from conda_build.jinja_context import context_processor
 
     path, filename = os.path.split(meta_path)
-    loaders = [jinja2.PackageLoader('conda_build'),
+    loaders = [# search relative to '<conda_root>/Lib/site-packages/conda_build/templates'
+               jinja2.PackageLoader('conda_build'),
+               # search relative to RECIPE_DIR
                jinja2.FileSystemLoader(path)
                ]
+    
+    # search relative to current conda environment directory
+    conda_env_path = os.environ.get('CONDA_ENV_PATH',  # path to current conda environment
```

Comment:
When I call `activate foo`, conda automatically defines `CONDA_ENV_PATH=<conda_root>/envs/foo`. Is this non-portable?

Using the `PREFIX` doesn't work because the `_build` directory is not yet populated when `meta.yaml` is parsed, so one can't find any file there. Python's `sys.prefix` is unsuitable as well because the template mechanism should also work for non-python builds (such as `libtiff`).